### PR TITLE
[WIP] Allow built instance as bulkCreate argument

### PR DESCRIPTION
**Pull request by phanect**
This PR allows users to give built instances as `bulkCreate` argument. It works as if it is "`bulkSave`"

Example:

```javascript
var users = [];
users.push(sequelize.models.User.build({
  username: "Alice",
  password: "foobar"
}));

users.push(sequelize.models.User.build({
  username: "Bob",
  password: "tetete"
}));

sequelize.models.User.bulkCreate(users);

```

This PR is not completed yet. (requires docs)

### CODE REVIEW

**Comment by phanect** [comment]
I want to hear the opinion about API design.
I considered to create `bulkSave` instead of allowing instance for `bulkCreate`, but currently I didn't. Which is better to create `bulkSave`, or keep current implementation?

**Comment by phanect** [comment]
I want to hear the opinion about API design.
I considered to create bulkSave instead of allowing instance for bulkCreate, but currently I didn't. Which is better to create bulkSave, or keep current implementation?

**Comment by mickhansen** [comment]
What's the reasoning for this exactly? What does it offer over just using an array of values?

File: test/integration/model.test.js [hunk 1]
```diff
@@ -2664,7 +2664,40 @@ describe(Support.getTestDialectTeaser('Model'), function() {
     return this.sequelize.sync({force: true});
   });
 
-  describe('bulkCreate errors', function() {
+  describe('bulkCreate', function() {
```
Comment (mickhansen) [inline test/integration/model.test.js]: Probably wanna place this test other than where we test for errors.

**Comment by phanect** [comment]
@mickhansen 

Because I want to use `set[AS]` and `add[AS]`.

Example:
```
var members = [];

var org = sequelize.models.Organization.build({
  name: "xx Foundation"
});

members.push(sequelize.models.Member.build({
  name: "Alice"
}));

members.push(sequelize.models.Member.build({
  name: "Bob"
}));

members.forEach(function(member) {
  org.addMember(member, { save: false });
});

sequelize.models.User.bulkCreate(members);
```

Currently I can do the same thing by adding `organization_id` as follows:
```
{
  name: "Bob",
  organization_id: org.get("id")
}
```
But this way is not clean and might cause bug.

File: test/integration/model.test.js (hunk 1 above)
Comment (phanect) [inline test/integration/model.test.js]: OK.
Where is the best place?
Is it OK to create another `describe` at the same level as `bulkCreate errors` in test/integration/model.test.js?

Comment (mickhansen) [inline test/integration/model.test.js]: https://github.com/sequelize/sequelize/blob/master/test/integration/model/create.test.js#L1288

**Comment by mickhansen** [comment]
Makes sense!

**Comment by mickhansen** [comment]
Although i believe we only support `save: false` for belongsTo so you might want to use the setter inversely for 1:M (i.e. add to the belongsTo source).

File: test/integration/model.test.js (hunk 1 above)
Comment (phanect) [inline test/integration/model.test.js]: Oh, missed it! OK.

**Comment by phanect** [comment]
I will restart working for this at 16th or 17th ;)

**Comment by phanect** [comment]
It seems to require heavy modification to realize association before `save()`. (example shown in  [7252891](https://github.com/phanect/sequelize/commit/769894e6b3eaaf1ec39a5c0e154e34362576940b))

[When I run `setOrganization()`](https://github.com/phanect/sequelize/commit/769894e6b3eaaf1ec39a5c0e154e34362576940b#diff-4b224bbc09dfb867f8671d77e298bb8cR1446), `org.id` is `null` and then `null` is assigned to `members[*].OrganizationId`. So it is difficult to determine which `Organization` and `Member`s are associated.

I terminate to work for this and will close this PR...

**Comment by mickhansen** [comment]
Yeah you only have access to the ID post bulkCreate when using `returning: true` with postgres. mysql and sqlite does not support returning multiple insert id's.

**Pull request closed without merging**